src/main/java/util/game/TransparentRisk.java

```diff
@@ -111,4 +115,35 @@ private String createContinentRateOutput(Map<Integer, Double> occupancyRateMap)
         }
         return sb.toString();
     }
+
+    private String toReadableAction(RiskAction action, RiskBoard board) {
+        if (action == null) {
+            return "Start";
+        }
+        if (action.attackingId() == -2) {
+            return "End of phase";
+        }
+        if (action.defendingId() == -3) {
+            //??
+            return String.format("Play cards for bonus (%d) - %s", action.troops(), action.toString());
+        }
+        if (action.defendingId() == -4) {
+            return String.format("Receive bonus (%d)", action.troops());
+        }
+        if (board.isReinforcementPhase()) {
+            return String.format("Reinforce %s (%d)", TerritoryNames.of(action.defendingId()), action.troops());
+        } else if (board.isAttackPhase()) {
+            if (action.attackingId() == -1) {
+                return String.format("Attack lost %d, Defence lost %d", action.attackerCasualties(), action.defenderCasualties());
+            }
+            return String.format("Attack %s -> %s (%d)", TerritoryNames.of(action.attackingId()),
+                    TerritoryNames.of(action.defendingId()), action.troops());
+        } else if (board.isOccupyPhase()) {
+            return String.format("Occupying with %d", action.troops());
```

Comment:
Could you maybe add the country being fortified here as well?

Reply:
occupy != fortify != reinforcement
The occupy action has no reference to the country afaik and always depends on the previous action (the successfull attack)